clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -547,7 +548,7 @@ public Long append(SimpleRecord record) {
      * @param value The control record value
      * @return CRC of the record or null if record-level CRC is not supported for the message format
      */
-    private Long appendControlRecord(long timestamp, ControlRecordType type, ByteBuffer value) {
+    public Long appendControlRecord(long timestamp, ControlRecordType type, ByteBuffer value) {
```

Comment:
Not really want to make this public, but at least we should avoid deserialize the data and serialize it again... in MockLog

Reply:
Are you saying it's an optimization for MockLog? Not sure that is worthwhile.

Reply:
Removed